src/components/Markdown/index.jsx

```diff
@@ -0,0 +1,219 @@
+      overflow: 'auto',
+    },
+    '& code': {
+      display: 'inline-block',
+      lineHeight: 1.6,
+      fontFamily: 'Consolas, "Liberation Mono", Menlo, Courier, monospace',
+      padding: '3px 6px',
+      color: theme.palette.text.primary,
+      backgroundColor: theme.palette.background.paper,
+      fontSize: 14,
+    },
+    '& p code, & ul code, & pre code': {
+      fontSize: 14,
+      lineHeight: 1.6,
+    },
+    '& h1': {
+      ...theme.typography.display2,
+      color: theme.palette.text.secondary,
+      margin: '0.7em 0',
+    },
+    '& h2': {
+      ...theme.typography.display1,
+      color: theme.palette.text.secondary,
+      margin: '1em 0 0.7em',
+    },
+    '& h3': {
+      ...theme.typography.headline,
+      color: theme.palette.text.secondary,
+      margin: '1em 0 0.7em',
+    },
+    '& h4': {
+      ...theme.typography.title,
+      color: theme.palette.text.secondary,
+      margin: '1em 0 0.7em',
+    },
+    '& p, & ul, & ol': {
+      lineHeight: 1.6,
+    },
+    '& h1, & h2, & h3, & h4': {
+      '& code': {
+        fontSize: 'inherit',
+        lineHeight: 'inherit',
+        // Remove scroll on small screens.
+        wordBreak: 'break-word',
+      },
+      '& .anchor-link-style': {
+        opacity: 0,
+        // To prevent the link to get the focus.
+        display: 'none',
+      },
+      '&:hover .anchor-link-style': {
+        display: 'inline-block',
+        opacity: 1,
+        padding: `0 ${theme.spacing.unit}px`,
+        color: theme.palette.text.hint,
+        '&:hover': {
+          color: theme.palette.text.secondary,
+        },
+        '& svg': {
+          width: '0.55em',
+          height: '0.55em',
+          fill: 'currentColor',
+        },
+      },
+    },
+    '& table': {
+      width: '100%',
+      display: 'block',
+      overflowX: 'auto',
+      borderCollapse: 'collapse',
+      borderSpacing: 0,
+      overflow: 'hidden',
+      '& .prop-name': {
+        fontSize: 13,
+        fontFamily: 'Consolas, "Liberation Mono", Menlo, monospace',
+      },
+      '& .required': {
+        color: theme.palette.type === 'light' ? '#006500' : '#9bc89b',
+      },
+      '& .prop-type': {
+        fontSize: 13,
+        fontFamily: 'Consolas, "Liberation Mono", Menlo, monospace',
+        color: theme.palette.type === 'light' ? '#932981' : '#dbb0d0',
+      },
+      '& .prop-default': {
+        fontSize: 13,
+        fontFamily: 'Consolas, "Liberation Mono", Menlo, monospace',
+        borderBottom: `1px dotted ${theme.palette.text.hint}`,
+      },
+    },
+    '& thead': {
+      fontSize: 14,
+      fontWeight: theme.typography.fontWeightMedium,
+      color: theme.palette.text.secondary,
+    },
+    '& tbody': {
+      fontSize: 14,
+      lineHeight: 1.5,
+      color: theme.palette.text.primary,
+    },
+    '& td': {
+      borderBottom: `1px solid ${theme.palette.divider}`,
+      padding: `${theme.spacing.unit}px ${theme.spacing.double}px ${
```

Comment:
If we are going to rely on non-standard theme properties, then we need a way for component consumers to create a `MuiThemeProvider` that does this for them.